component/core/src/main/java/org/exoplatform/social/core/jpa/storage/RDBMSActivityStorageImpl.java

```diff
@@ -1397,9 +1397,7 @@ public void updateActivity(ExoSocialActivity existingActivity) {
       if (existingActivity.getPermaLink() != null)
         updatedActivity.setPermaLink(existingActivity.getPermaLink());
       if (existingActivity.getTemplateParams() != null) {
-        Map<String, String> updatedActivityTemplateParams = updatedActivity.getTemplateParams();
-        updatedActivityTemplateParams.putAll(existingActivity.getTemplateParams());
-        updatedActivity.setTemplateParams(updatedActivityTemplateParams);
+        updatedActivity.setTemplateParams(existingActivity.getTemplateParams());
```

Comment:
normally we should keep the default code, otherwises we will have regressions

Reply:
we haven't implemented odd things, it is well explained in the commit. The test case for update activity already exists but not for this specific case maybe.

Reply:
my point is exactly that. This regression should not have been possible if we had had a unit test that would protect this case. It should be added because this code is not self explanatory.